### DISCUSSION TITLE
How do I use shardingkey in async mode

### DISCUSSION THREAD

**Discussion by apple-yzhang** [oracle/python-oracledb#317] [Q&A]
It seems the shardingkey is only allowed in thick mode, but async doesn't support thick mode.

Is shardingkey going to be supported in the async thin or thick mode in the near future?

**Comment by anthony-tuininga** [comment]
Sharding is not supported in thin mode yet. There are no plans to use thick mode in asyncio (significantly reduced performance) but I have created an enhancement request for adding sharding support to thin mode. I will discuss internally about priority.

**Reply by apple-yzhang** [reply]
Great! Do you have some performance benchmark numbers of using a shard vs using the thin mode? Currently, we have 4 shards, and assume data are distributed evenly, no hot zones etc. I'm thinking if should I move forward with thin mode now, or wait for the sharding support in thin mode.

**Reply by cjbj** [reply]
The two are very different technologies, so a comparison isn't helpful. Since it harder to change data layout, I would decide the schema first.
